service/azureconfig/v1/resource/vnetpeering/update.go

```diff
@@ -0,0 +1,143 @@
+	if err != nil {
+		return nil, microerror.Maskf(err, "NewUpdatePatch")
+	}
+	c, err := toVnetPeering(current)
+	if err != nil {
+		return nil, microerror.Maskf(err, "NewUpdatePatch")
+	}
+	d, err := toVnetPeering(desired)
+	if err != nil {
+		return nil, microerror.Maskf(err, "NewUpdatePatch")
+	}
+
+	patch, err := r.newUpdatePatch(ctx, a, c, d)
+	if err != nil {
+		return nil, microerror.Maskf(err, "NewUpdatePatch")
+	}
+
+	return patch, nil
+}
+
+func (r Resource) newUpdatePatch(ctx context.Context, azureConfig providerv1alpha1.AzureConfig, current, desired network.VirtualNetworkPeering) (*framework.Patch, error) {
+	patch := framework.NewPatch()
+
+	change, err := r.newUpdateChange(ctx, azureConfig, current, desired)
+	if err != nil {
+		return nil, microerror.Maskf(err, "newUpdatePatch")
+	}
+
+	patch.SetUpdateChange(change)
+
+	return patch, nil
+}
+
+func (r Resource) newUpdateChange(ctx context.Context, azureConfig providerv1alpha1.AzureConfig, current, desired network.VirtualNetworkPeering) (network.VirtualNetworkPeering, error) {
+	var change network.VirtualNetworkPeering
+
+	ok, err := needUpdate(current, desired)
+	if err != nil {
+		return change, microerror.Maskf(err, "newUpdateChange")
+	}
+	if ok {
+		change = desired
+	}
+
+	return change, nil
+}
+
+// needUpdate determine if current needs to be updated in order to comply with desired.
+// Following properties are compared (and must be present in desired)
+//     Name
+//     VirtualNetworkPeeringPropertiesFormat.AllowVirtualNetworkAccess
+//     VirtualNetworkPeeringPropertiesFormat.RemoteVirtualNetwork.ID
+func needUpdate(current, desired network.VirtualNetworkPeering) (bool, error) {
+	if desired.Name == nil ||
+		desired.VirtualNetworkPeeringPropertiesFormat == nil ||
+		desired.VirtualNetworkPeeringPropertiesFormat.AllowVirtualNetworkAccess == nil ||
+		desired.VirtualNetworkPeeringPropertiesFormat.RemoteVirtualNetwork == nil ||
+		desired.VirtualNetworkPeeringPropertiesFormat.RemoteVirtualNetwork.ID == nil {
+		return false, microerror.Maskf(invalidDesiredState, "got %+#v", desired)
+	}
+
+	if current.Name == nil || *current.Name != *desired.Name {
+		return true, nil
+	}
+
+	if current.VirtualNetworkPeeringPropertiesFormat == nil {
+		return true, nil
+	}
+
+	if current.VirtualNetworkPeeringPropertiesFormat.AllowVirtualNetworkAccess == nil ||
+		*current.VirtualNetworkPeeringPropertiesFormat.AllowVirtualNetworkAccess != *desired.VirtualNetworkPeeringPropertiesFormat.AllowVirtualNetworkAccess {
+		return true, nil
+	}
+
+	if current.VirtualNetworkPeeringPropertiesFormat.RemoteVirtualNetwork == nil ||
+		current.VirtualNetworkPeeringPropertiesFormat.RemoteVirtualNetwork.ID == nil ||
+		*current.VirtualNetworkPeeringPropertiesFormat.RemoteVirtualNetwork.ID != *desired.VirtualNetworkPeeringPropertiesFormat.RemoteVirtualNetwork.ID {
+		return true, nil
+	}
+
+	if current.VirtualNetworkPeeringPropertiesFormat.PeeringState == network.Disconnected {
```

Comment:
Handle case where peering is in `disconnected` state and need to be updated in order to be in `initiated` state and ready for a new connection.